File: perl/plus_operator/plus_operator/src/plus_operator/plus_operator.pl
#!/usr/bin/perl

# 数値加算
$x = 10;
$y = 20;
$ret = $x + $y;
print "$ret\n";

# 数字文字列なら数値に変換される.
$x1 = "10";
$x2 = "20";
$x = $x1 + $x2;
print "$x\n";

# ダブルクォート関係なく文字列でも数値として加算しようとする.
$s1 = "1ABC";
$s2 = "2DEF";
$str = $s1 + $s2;
print "$str\n";

File: cpp/plus_operator/plus_operator_4/src/plus_operator/plus_operator.cpp
// ヘッダのインクルード
#include <iostream> // 標準入出力
#include <string> // C++文字列

// main関数
int main(){

  // オブジェクトの宣言と初期化.
  std::basic_string<char> str1 = "DEF"; // std::basic_string<char>オブジェクトstr1を"DEF"で初期化.
  std::basic_string<char> ret; // 結果を格納するstd::basic_string<char>オブジェクトret.

  // char型ポインタの文字列とstd::basic_string<char>オブジェクトの連結.
  ret = "ABC" + str1; // "ABC"とstr1を連結し, retに格納.

  // 連結した文字列を出力.
  std::cout << ret << std::endl; // retを出力.

  // プログラムの終了
  return 0; // 0を返して正常終了.

}
